lib/functions/aeminitiateupload.js

```diff
@@ -104,11 +106,27 @@ class AEMInitiateUpload extends AsyncGeneratorFunction {
             const completeUrl = new URL(initiateResponse.completeURI, folderUrl);
             for (let i = 0; i < assets.length; ++i) {
                 const transferAsset = assets[i];
-                const { minPartSize, maxPartSize, uploadURIs, uploadToken } = files[i];
+                const { minPartSize, maxPartSize, uploadURIs, mimeType, uploadToken } = files[i];
+
                 // uploadToken is opaque to us, we just need to send it back with the completeUpload
                 if (!Number.isFinite(minPartSize) || !Number.isFinite(maxPartSize) || !Array.isArray(uploadURIs) || (uploadURIs.length === 0)) {
                     throw Error(`invalid multi-part information for ${transferAsset.target.url}: ${JSON.stringify(files[i])}`);
                 }
+
+                // if the client has no mimetype specified, use the mimetype provided by AEM, if AEM
+                // cannot detect a mimetype default to application/octet-stream
+                if (!transferAsset.metadata.contentType) {
+                    let contentType = mimeType;
+                    if (typeof contentType !== "string") {
```

Comment:
The type in the contentype change from `lib/functions/aemcompleteupload.js` is not checked there. SHould it be verified it's a string there too?

Reply:
It's here because it's checking the server response. I changed it to be more explicit and added more checks, but only on type though. Might be good to get that expanded later, but I don't want to risk rejecting a response that is ok but technically invalid.